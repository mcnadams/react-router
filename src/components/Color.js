import React from 'react';
import PropTypes from 'prop-types';

function Color({ match }) {
  const style = {
    backgroundColor: decodeURIComponent(match.params.color),
    width: '250px',
    height: '250px'
  };

  return <div style={style}></div>;
}

Color.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      color: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Color;
