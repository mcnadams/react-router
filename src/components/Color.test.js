import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color', () => {
  it('Color', () => {
    const match = {
      params: {
        color: 'pink'
      }
    };
    const wrapper = shallow(<Color match={match} />);
    expect(wrapper).toMatchSnapshot();
  });
});
