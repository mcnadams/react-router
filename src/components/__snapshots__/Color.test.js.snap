// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Color 1`] = `
<div
  style={
    Object {
      "backgroundColor": "pink",
      "height": "250px",
      "width": "250px",
    }
  }
/>
`;
